File: ArrayAndString/119. Pascal's Triangle II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex==0:
            return [1]
        elif rowIndex==1:
            return [1,1]
        
        result=[[1],[1,1]]
        
        
        for i in range(1,rowIndex):
            generate=[0]*(len(result[i])+1)
            
            generate[0],generate[-1]=1,1
            for j in range(1,len(generate)-1):
                generate[j]=result[i][j-1]+result[i][j]
            result.append(generate)
            
        return result[-1]
        
        """
        #rework old version
        allTheLists = []
        for rowNum in range(rowIndex+1):
            allTheLists.append([])
        allTheLists[0].append(1) #first row

        for rowNumbers in range(1,rowIndex+1):
            allTheLists[rowNumbers].append(1) #the first one
            for colNumbers in range(1,len(allTheLists[rowNumbers-1])):
                allTheLists[rowNumbers].append(allTheLists[rowNumbers-1][colNumbers-1]+allTheLists[rowNumbers-1][colNumbers])
            allTheLists[rowNumbers].append(1) #the last one at the end
        return allTheLists[rowIndex]
        """



        

File: ArrayAndString/119. Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        int[] prev = new int[rowIndex + 1];
        int[] curr = new int[rowIndex + 1];
        prev[0] = 1; curr[0] = 1;
        for(int i = 1, mid; i <= rowIndex ; i++){
            mid = i / 2 + 1;
            curr[i] = 1;
            for(int j = 1; j < mid; j++){
                curr[j] = prev[j] + prev[j - 1];
                curr[i - j] = curr[j];
            }
            int[] temp = curr;
            curr = prev;
            prev = temp;
        }
        List<Integer> Curr = new ArrayList<Integer>();
        for(int c : prev) Curr.add(c);
        return Curr;
    }

        /*
        if(rowIndex==0) return [1];
        if(rowIndex==1) return [1,1];
        List<Integer> prev = new ArrayList<Integer>();
        prev = {1,1};
        List<Integer> curr = new ArrayList<Integer>();
        for(int i = 1; i < rowIndex; i++){
            curr.add(1);
            for(int first = 0; first < prev.size()-1; first++){
                curr.add(prev.get(first)+prev.get(first+1));
            }
            curr.add(1);
            prev = curr;
        }
        return curr;
        */
}